### PR TITLE
NXP-26154: check in session if id and password fields are prefixed

### PR DESCRIPTION
**Pull request by funshodavid**

### CODE REVIEW

**Comment by funshodavid** [comment]
T&P failing on unrelated issue : https://qa2.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-fdavid-9.10/34/

**Comment by kevinleturc** [comment]
👍

**Comment by pierre-gautier** [comment]
👍

File: nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-mongodb/src/main/java/org/nuxeo/directory/mongodb/MongoDBSession.java [hunk 1]
```diff
@@ -499,4 +507,20 @@ protected boolean checkEntryTenantId(String entryTenantId) {
...
 
+    protected String getPrefixedIdField() {
+        Field idField = directory.getSchemaFieldMap().get(getIdField());
+        if (idField == null) {
+            return null;
+        }
+        return idField.getName().getPrefixedName();
+    }
+
+    protected String getPrefixedPasswordField() {
```
Comment (efge) [inline nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-mongodb/src/main/java/org/nuxeo/directory/mongodb/MongoDBSession.java]: You could have factored out the body of `getPrefixedIdField` and `getPrefixedPasswordField` which are very similar.

**Pull request merged**